### PR TITLE
Add a linter to ensure controllers inherit from application controller

### PR DESCRIPTION
**Pull request by brentgaither**
We want to make sure that all of our controllers are inheriting from application controller which ensures that they are logged in and a few more essential things to use mavenlink.

### CODE REVIEW

**Comment by brandonduff** [comment]
We'll probably also want to make sure any controllers in the `Api` module inherit from `ApiController`.

File: lib/rubocop/cop/mavenlint/use_application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      #
+      # should be
+      #
+      #   class SomeController < ApplicationController
+      #   end
+      #
+      class UseApplicationController < RuboCop::Cop::Cop
+        MSG = 'Controllers should subclass `ApplicationController`.'.freeze
+        SUPERCLASS = 'ApplicationController'.freeze
+        BASE_PATTERN = '(const (const nil? :ApplicationController) :Base)'.freeze
```
Comment (ahuth) [inline lib/rubocop/cop/mavenlint/use_application_controller.rb]: Should this be:

```lisp
(const (const nil? :ActionController) :Base)
```

?

A test could help give us confidence

Comment (brentgaither) [reply lib/rubocop/cop/mavenlint/use_application_controller.rb]: Thanks for always keeping us on track! I added some tests around it, and fixed the mistakes 😄

Comment (brentgaither) [reply lib/rubocop/cop/mavenlint/use_application_controller.rb]: @ahuth  It looks like you are the owner on ruby gems, could you add me? 
`gem owner mavenlint --add brent.w.gaither@gmail.com`

File: lib/rubocop/cop/mavenlint/use_application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      #
+      #   class SomeController < ActionController::Base
+      #   end
+      #
+      # should be
+      #
+      #   class SomeController < ApplicationController
+      #   end
+      #
+      class UseApplicationController < RuboCop::Cop::Cop
```
Comment (shirish-pampoorickal) [inline lib/rubocop/cop/mavenlint/use_application_controller.rb]: CRITICAL: Do we need to enforce this linting rule in rubocop.yml?

Comment (brentgaither) [reply lib/rubocop/cop/mavenlint/use_application_controller.rb]: Thanks for catching that, its added now!

File: lib/rubocop/cop/mavenlint/use_application_controller.rb (hunk 1 above)
Comment (ahuth) [reply lib/rubocop/cop/mavenlint/use_application_controller.rb]: You're now an owner!

**Pull request merged**